tla/install_deps.py

```diff
@@ -0,0 +1,149 @@
+    assert os.path.exists(file_path)
+    bin_path = None
+
+    if file_name.endswith(".bin"):
+        os.chmod(file_path, os.stat(file_path).st_mode | stat.S_IEXEC)
+        subprocess.Popen(f"{file_path} -d {dest}".split()).wait()
+        bin_path = f"{dest}/bin"
+
+    elif file_name.endswith(".tgz"):
+        with tarfile.open(f"/tmp/{file_name}") as tar:
+            tar.extractall(dest)
+            rel_bin_path = next(
+                member.name for member in tar.getmembers() if "bin" in member.name)
+            bin_path = os.path.join(dest, rel_bin_path)
+
+    elif file_name.endswith(".jar"):
+        shutil.copyfile(file_path, os.path.join(dest, file_name))
+
+    if bin_path is not None:
+        append_bashrc(f"export PATH:$PATH:{bin_path}")
+
+
+def _parse_args() -> argparse.Namespace:
+
+    parser = argparse.ArgumentParser(
+        description="Install CCF TLA+ dependencies",
+    )
+
+    parser.add_argument(
+        "--tlaps",
+        action="store_true",
+        default=False,
+    )
+
+    parser.add_argument(
+        "--apalache",
+        action="store_true",
+        default=False,
+    )
+
+    parser.add_argument(
+        "--skip-apt-packages",
+        action="store_false",
+        default=True,
+        dest="apt_packages"
+    )
+
+    return parser.parse_args()
+
+
+def install_tlc():
+
+    java = "java -XX:+UseParallelGC"
+    tlaplus_path = "~/.vscode-remote/extensions/alygin.vscode-tlaplus-nightly-*/tools/tla2tools.jar"
+    tlcmax_args = "-Dtlc2.tool.fp.FPSet.impl=tlc2.tool.fp.OffHeapDiskFPSet -Dtlc2.tool.ModelChecker.BAQueue=true"
+    copy_tlaplus = f"-cp {tlaplus_path} tlc2.TLC"
+
+    def get_run_tlc_max(size: int) -> str:
+        return f"{java} -XX:MaxDirectMemorySize={size}g {tlcmax_args} {copy_tlaplus}"
+
+    set_alias("tlcrepl", f"{java} -cp {tlaplus_path} tlc2.REPL")
+    set_alias("tlc", f"{java} {copy_tlaplus}")
+    set_alias("tlcmax4", get_run_tlc_max(4))
+    set_alias("tlcmax8", get_run_tlc_max(8))
+    set_alias("tlcmax16", get_run_tlc_max(16))
+    set_alias("tlcmax32", get_run_tlc_max(32))
+
+
+def install_deps(args: argparse.Namespace):
+
+    # Setup tools directory
+    tools_dir = os.path.join(TLA_DIR, "tools")
+    def create_tools_dir():
+        if not os.path.exists(tools_dir):
+            os.mkdir(tools_dir)
+
+    install_tlc()
+
+    if args.tlaps:
+        create_tools_dir()
+        fetch_latest(
+            url="https://github.com/tlaplus/tlapm/releases/download/v1.4.5/tlaps-1.4.5-x86_64-linux-gnu-inst.bin",
+            dest=os.path.join(tools_dir, "tlaps"),
+        )
+
+    if args.apalache:
+        create_tools_dir()
+        fetch_latest(
+            url="https://github.com/informalsystems/apalache/releases/latest/download/apalache.tgz",
+            dest=tools_dir,
+        )
+
+    if args.apt_packages:
+        subprocess.Popen(
+            "sudo apt-get install -y --no-install-recommends".split() + [
+                "graphviz",
```

Comment:
Things like these that aren't upgraded frequently (ie. unlike the nightly tla build) are best put in the base image eventually, rather than installed every time.

Reply:
Yeah makes sense, issue is here https://github.com/microsoft/CCF/issues/4532